app/src/pages/BinaryMetricsPage/BinaryMetricsPage.tsx

```diff
@@ -65,6 +65,27 @@ const mapStateToProps = (state: Store): BinaryMetricsDispatchProps => ({
       objKey: aTitle,
     })
   ),
+  confusionMatrix: {
+    totalCount: Math.pow(
+      state.DatasetsStore.selectedDataset?.numberOfRecords ?? 0,
+      2
+    ),
+    falseNegatives: state.MetricsStore.falseNegatives?.data.length ?? undefined,
+    falsePositives: state.MetricsStore.falsePositives?.data.length ?? undefined,
+    trueNegatives:
+      state.MetricsStore.falseNegatives !== undefined &&
+      state.MetricsStore.falsePositives !== undefined &&
+      state.MetricsStore.truePositives?.data.length !== undefined
+        ? Math.pow(
+            state.DatasetsStore.selectedDataset?.numberOfRecords ?? 0,
+            2
+          ) -
+          ((state.MetricsStore.falseNegatives?.data.length ?? 0) +
+            (state.MetricsStore.falsePositives?.data.length ?? 0) +
+            (state.MetricsStore.truePositives?.data.length ?? 0))
+        : undefined,
+    truePositives: state.MetricsStore.truePositives?.data.length ?? undefined,
```

Comment:
I think a separate function that calculates the confusion matrix is good because it is not easy to follow the structure of undefined checks. So a function checks if one value is  undefined and if not so, it calculates the confusion matrix straigth forward.

Reply:
Will be resolved by a future PR that @florian-papsdorf is currently working on.